src/App.js

```diff
@@ -2,24 +2,32 @@ import React from "react";
 import './App.css';
 import Header from "./Header";
 import Movies from "./Movies";
-import movieData from './data';
 import SingleMovieCard from "./SingleMovieCard";
+import {allMoviesData, singleMovieData} from './APIcalls';
+
 
 class App extends React.Component {
   constructor() {
     super()
     this.state = {
-      allMovies: movieData.movies,
+      allMovies: [],
       isSingleMovie: false,
-      singleMovie: ''
+      singleMovie: '',
+      // singleMovieOverview: '',
+      error: ''
     }
   }
 
-  displayMovieInfo = (id) => {
-  const singleMovieDetails = this.state.allMovies.find((movie) => {
-    return movie.id === id;
-  })
-    this.setState({ isSingleMovie: true, singleMovie: singleMovieDetails })
+  componentDidMount = () => {
+    return allMoviesData()
+    .then(data => this.setState({allMovies: data.movies}))
+    .catch(error => this.setState({error: 'Oops! Something went wrong!'}))
+  }
+
+
+displayMovieInfo = (id) => {
+  return singleMovieData(id)
+  .then(data => this.setState({isSingleMovie: true, singleMovie: data.movie}))
```

Comment:
Be sure to use `catch` at the end of any `.then()` chain to make sure you're capturing any errors that might occur.